server/pbenchinacan/load_keycloak.sh

```diff
@@ -17,12 +17,53 @@
 # "http://localhost:8080/*" unless specified otherwise by 'KEYCLOAK_REDIRECT_URI'
 # env variable.
 
+KEYCLOAK_BASE_IMAGE=${KEYCLOAK_BASE_IMAGE:-"quay.io/keycloak/keycloak:latest"}
 KEYCLOAK_HOST_PORT=${KEYCLOAK_HOST_PORT:-"http://localhost:8090"}
+KEYCLOAK_HEALTH_URI="${KEYCLOAK_HOST_PORT}/health"
 KEYCLOAK_REDIRECT_URI=${KEYCLOAK_REDIRECT_URI:-"http://localhost:8080/*"}
 ADMIN_USERNAME=${ADMIN_USERNAME:-"admin"}
 ADMIN_PASSWORD=${ADMIN_PASSWORD:-"admin"}
 REALM=${KEYCLOAK_REALM:-"pbench"}
 CLIENT=${KEYCLOAK_CLIENT:-"pbench-server"}
+KEYCLOAK_IMAGE_TAG=${KEYCLOAK_IMAGE_TAG:-"latest"}
+
+container_name="mykeycloak"
+podman run -d --rm --name ${container_name} -p 8090:8090 \
+-e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin \
+${KEYCLOAK_BASE_IMAGE} start-dev --health-enabled=true --http-port=8090
+
+end_in_epoch_secs=$(( $(date +"%s") + 120 ))
+
+until curl -s -o /dev/null ${KEYCLOAK_HEALTH_URI}; do
+  if [[ $(date +"%s") -ge ${end_in_epoch_secs} ]]; then
+    echo "Timed out connecting to Keycloak" >&2
+    exit 1
+  fi
+  echo "Waiting for the Keycloak server" >&2
+  sleep 2
+done
+
+echo "Keycloak health url is up" >&2
+
+status_code=$(curl -s -o /dev/null -w "%{http_code}" ${KEYCLOAK_HEALTH_URI})
+
+while [[ $(date +"%s") -le ${end_in_epoch_secs} ]]; do
+  if [[ "${status_code}" == "200" ]]; then
+    keycloak_status=$(curl -s ${KEYCLOAK_HEALTH_URI} | jq -r ".status")
+    if [[ ${keycloak_status} == "UP" ]]; then
+      echo "Keycloak Server is up" >&2
+      break
+    else
+      echo "Keycloak Server status is not UP" >&2
+    fi
+  else
+    echo "Keycloak health status code ${status_code}" >&2
+  fi
+  status_code=$(curl -s -o /dev/null -w "%{http_code}" ${KEYCLOAK_HEALTH_URI})
+  sleep 2
+done
```

Comment:
There's no need/point to sleeping between the `curl` invocation and the checks -- that potentially just wastes time.  Moreover, if you do the `curl` invocation at the top of the loop, then you don't need to have the "prime-the-pump" statement before the loop:
```
while [[ $(date +%s) -le ${end_in_epoch_secs} ]]; do
  status_code=$(curl -s -o /dev/null -w "%{http_code}" ${KEYCLOAK_HEALTH_URI})
  if [[ "${status_code}" == "200" ]]; then
    keycloak_status=$(curl -s ${KEYCLOAK_HEALTH_URI} | jq -r ".status")
    if [[ ${keycloak_status} == "UP" ]]; then
      echo "Keycloak Server is up" >&2
      break
    else
      echo "Keycloak Server status is not UP" >&2
    fi
  else
    echo "Keycloak health status code ${status_code}" >&2
  fi
  sleep 2
done
```
It's a shame to have to make two separate `curl` requests, but it saves us from having to use the local filesystem, so maybe it's the lesser of evils.

Reply:
removing the sleep just prints lot of `Keycloak Server status is not UP` statements. If I remove the sleep then I should also remove the `echo` statement. However, having that echo statement is sort of assuring that it is trying to get the health-check which is why I wanted to keep without cluttering the stdout.

Reply:
Oh, no, don't remove the sleep!

Just move the `curl` from the bottom to the top of the loop.  (See code above.)